tests/functional/test_endpoint_rulesets.py

```diff
@@ -305,3 +306,28 @@ def builtin_overwriter_handler(builtins, **kwargs):
             except (ClientError, ResponseParserError):
                 pass
         assert len(http_stubber.requests) == 0
+
+
+@mock.patch('botocore.signers.RequestSigner.region_name', 'us-east-1')
+@mock.patch(
+    'botocore.regions.EndpointRulesetResolver.auth_schemes_to_signing_ctx',
+    return_value=('v4', {'region': 'us-west-2'}),
+)
+@mock.patch(
+    'botocore.regions.EndpointRulesetResolver.uses_builtin_data_path', False
+)
+@mock.patch(
+    'botocore.regions.EndpointRulesetResolver.credential_scope_set', True
+)
+def test_signing_region_mismatch_warns(auth_schemes_mock, patched_session):
```

Comment:
It's unfortunate we have to mock so much out, but this is an edge case and would require a ton of scaffolding to be able to handle a warning. We could do this with a mock ruleset instead, but would need to mock out the function that returns the ruleset data to the client. The uses builtin check is already mocked for other behavior [here](https://github.com/boto/botocore/blob/7c1e00cd2bb38a3d41f1ca6849decfbe30605d56/tests/unit/test_args.py#L84) and would be painful to set up. It also seems like too much effort to make a custom endpoints file that would also need to be mocked out.